daal4py/sklearn/test/test_run_to_run_stability_tests.py

```diff
@@ -14,322 +14,350 @@
-    name = ["transform", "result1.eigenvalues", "result1.eigenvectors", "result1.means", "result1.variances"]
-    return res, name
-
-##################################################
-
-class Test(unittest.TestCase):
-
-    #---------------------Passed---------------------
-
-    def test_knn_class_brute_uniform(self):
-        run_clf_test(run_knn_class_brute_uniform)
-
-    def test_knn_class_brute_distance(self):
-        run_clf_test(run_knn_class_brute_distance)
-
-    def test_knn_class_kdtree_uniform(self):
-        run_clf_test(run_knn_class_kdtree_uniform)
-
-    def test_knn_class_kdtree_distance(self):
-        run_clf_test(run_knn_class_kdtree_distance)
-
-    def test_dbascan(self):
-        run_clf_test(run_dbascan) 
-
-    def test_svc_linear(self):
-        run_clf_test(run_svc_linear)
-
-    def test_svc_rbf(self):
-        run_clf_test(run_svc_rbf)
-
-    def test_svc_rbf_const_gamma(self):
-        run_clf_test(run_svc_rbf_const_gamma)
-
-    #---------------------Failed---------------------
-
-    @unittest.skip("unstable")
-    def test_kmeans_plus_plus(self):
-        run_clf_test(run_kmeans_plus_plus)
-
-    @unittest.skip("unstable")
-    def test_kmeans_random(self):
-        run_clf_test(run_kmeans_random)
-
-    @unittest.skip("unstable")
-    def test_elasticnet_regression(self):
-        run_reg_test(run_elasticnet_regression)
-
-    @unittest.skip("unstable")
-    def test_lasso_regression(self):
-        run_reg_test(run_lasso_regression)
-
-    @unittest.skip("unstable")
-    def test_pca_full(self):
-        run_clf_test(run_pca_full)
-
-    @unittest.skip("unstable")
-    def test_pca_daal4py_corr(self):
-        run_clf_test(run_pca_daal4py_corr)
-
-    #---------------------Expected to be fixed in next release---------------------
-
-    @unittest.skip("unstable")
-    def test_log_regression_newton(self):
-        run_clf_test(run_log_regression_newton)
-
-    @unittest.skip("unstable")
-    def test_log_regression_lbfgs(self):
-        run_clf_test(run_log_regression_lbfgs)
-
-    @unittest.skip("unstable")
-    def test_log_regression_cv_newton(self):
-        run_clf_test(run_log_regression_cv_newton)
-
-    @unittest.skip("unstable")
-    def test_log_regression_cv_lbfgs(self):
-        run_clf_test(run_log_regression_cv_lbfgs)
-
-    @unittest.skip("unstable")
-    def test_linear_regression(self):
-        run_reg_test(run_linear_regression)
-
-    @unittest.skip("unstable")
-    def test_ridge_regression(self):
-        run_reg_test(run_ridge_regression)
-
-    @unittest.skip("unstable")
-    def test_rf_class(self):
-        run_clf_test(run_rf_class)
-
-    @unittest.skip("unstable")
-    def test_rf_reg(self):
-        run_reg_test(run_rf_regression)
-
 
-if __name__ == '__main__':
-    unittest.main()
+def _run_test(model, methods, dataset):
+    # ------------light datasets------------
+    if dataset in ['blobs', 'classifier', 'sparse']:
+        X, y = load_iris(return_X_y=True)

```

Comment:
There is a lot of copy-paste, this can be improved:

```python
datasets = []
if dataset in ['blobs', 'classifier', 'sparse']:
  X, y = load_iris(return_X_y=True)
  if dataset == 'sparse':
            X = sparse.csr_matrix(X)
  datasets.append((X,y))
   X, y = load_breast_cancer(return_X_y=True)
        if dataset == 'sparse':
            X = sparse.csr_matrix(X)
  datasets.append((X,y))
  ...
 # some for regression datasets
 ...
 for X, y in datasets:
    for i in range(10):
      ...



```